sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MutableFieldsShouldNotBe.cs

```diff
@@ -122,5 +120,17 @@ private static bool IsValidReadOnlyInitializer(ISymbol symbol, ExpressionSyntax
             return (methodSymbol != null && methodSymbol.ReturnType.DerivesOrImplementsAny(AllowedTypes)) ||
                    (equalsLiteral != null && equalsValue.IsKind(SyntaxKind.NullLiteralExpression));
         }
+
+        private static ISymbol GetFieldInitializerSymbol(ExpressionSyntax initializer, SemanticModel model)
+        {
+            if (initializer == null)
+            {
+                return null;
+            }
+
+            var symbolInfo = model.GetSymbolInfo(initializer);
+
+            return symbolInfo.Symbol ?? symbolInfo.CandidateSymbols.FirstOrDefault();

```

Comment:
For some reason (couldn't find it), `ImmutableArray` ctors can't resolve properly. I was tempted to remove this case but I prefer to have some FN rather than FP.